javalang/parser.py

```diff
@@ -1731,7 +1740,23 @@ def parse_expressionl(self):
             return tree.TernaryExpression(condition=expression_2,
                                           if_true=true_expression,
                                           if_false=false_expression)
-
+        if self.would_accept('->'):
+            body = self.parse_lambda_method_body()
+            return tree.LambdaExpression(parameters=[expression_2],
+                                         body=body)
+        if self.try_accept('::'):
+            if self.would_accept('<'):
```

Comment:
Correct I am not handling all the different cases as the test code documents with some failing test case.